src/main/java/ie/gmit/softwareeng/labexam/Playlist.java

```diff
@@ -1,71 +1,117 @@
 package ie.gmit.softwareeng.labexam;
 
+import java.util.*;
+
 public class Playlist {
+    private List<Song> songList = new ArrayList<>();
+    private String playlistName;
+    private int songIndex = 0;
+
     public Playlist(String playlistName) {
-        throw new UnsupportedOperationException("Not implemented yet");
+        this.playlistName = playlistName;
     }
 
     public String getName() {
-        throw new UnsupportedOperationException("Not implemented yet");
+        return playlistName;
     }
 
     public boolean contains(Song song) {
-        throw new UnsupportedOperationException("Not implemented yet");
+        return songList.stream().anyMatch(s -> s.equals(song));
```

Comment:
This works but you could have used songList.contains(song) instead. A bit less complicated